Sources/SwiftParser/Types.swift

```diff
@@ -793,11 +793,18 @@ extension Parser.Lookahead {
       return true
     }
 
-    if self.at(anyIn: EffectSpecifier.self) != nil {
+    if let effect = self.at(anyIn: EffectSpecifier.self) {
       if self.peek().rawTokenKind == .arrow {
         return true
       }
 
+      if effect.spec.isThrowsSpecifier && self.peek().rawTokenKind == .leftParen {
+        var backtrack = self.lookahead()
+        backtrack.consumeAnyToken()
+        backtrack.skipSingle()
+        return backtrack.atFunctionTypeArrow()
+      }
```

Comment:
I think this doesn’t handle `async throws(MyError)` because the current token is `async` here, which means that we end up in the `if` below, which just consumes `throws` but not `MyError`.

Reply:
The `consumeAnyToken()` consumes the `throws`, and the `skipSingle` consumes a parenthesized expression.